patch-db/src/model.rs

```diff
@@ -55,46 +101,93 @@ impl<T: Serialize + for<'de> Deserialize<'de>> DerefMut for ModelDataMut<T> {
         &mut self.current
     }
 }
-
 #[derive(Debug)]
 pub struct Model<T: Serialize + for<'de> Deserialize<'de>> {
-    ptr: JsonPointer,
+    pub(crate) path: ModelPaths,
```

Comment:
This file, is where I spent most of my time. This has a high chance of being a cause of issues?